code/__DEFINES/dcs/signals/signals_turf.dm

```diff
@@ -16,8 +16,12 @@
 #define COMSIG_TURF_ON_SHUTTLE_MOVE "turf_on_shuttle_move"
 ///from /turf/open/temperature_expose(datum/gas_mixture/air, exposed_temperature)
 #define COMSIG_TURF_EXPOSE "turf_expose"
-///from /turf/proc/immediate_calculate_adjacent_turfs()
-#define COMSIG_TURF_CALCULATED_ADJACENT_ATMOS "turf_calculated_adjacent_atmos"
+///from [/turf/proc/immediate_calculate_adjacent_turfs]: ([atmos_adjacent_turfs][/list])
+#define COMSIG_TURF_ADJACENT_ATMOS_RECALCULATING "turf_adjacent_atmos_recalculating"
+///from [/turf/proc/immediate_calculate_adjacent_turfs]: ([changed_turf][/turf], new_status, old_status)
+#define COMSIG_TURF_ADJACENT_ATMOS_CHANGED "turf_adjacent_atmos_changed"
+///from [/turf/proc/immediate_calculate_adjacent_turfs]: (diff_directions)
+#define COMSIG_TURF_ADJACENT_ATMOS_UPDATED "turf_adjacent_atmos_updated"
```

Comment:
What's the use of these three being separate?

Reply:
Recalculating is for eventual signal hooks to support portal atmos linking. The reason the other two are separate is because I don't think we're supposed to have more than one SEND_SIGNAL that sends the same signal.

Reply:
Also the args are slightly different.

Reply:
Might be able to unify the args, but that would require building a new list per signal call.

Reply:
I'm ok with having a before/after signal, but it's also fine to send the same one twice here, at least imo
Especially if they both mean the same thing. You're working with internal code here, it's optimizations shouldn't be reflected in the output that code relying on it sees

Reply:
Be careful with how much you add here. this code isn't particularly hot during runtime, but it's a good bit of init iirc